core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4652,7 +4652,7 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
    * Accesses the current scope and uses it to perform the specified effect.
    */
   def scopeWith[R, E, A](f: Scope => ZIO[R, E, A])(implicit trace: Trace): ZIO[R with Scope, E, A] =
-    ZIO.serviceWithZIO[Scope](f)
+    FiberRef.currentEnvironment.getWith(env => f(env.unsafe.getScope(Unsafe)))
```

Comment:
This is 'safe' because of `with Scope`?

Reply:
Yes. It's not any different than what was happening previously. The only thing that is making things "safe" is that the Scope is that `Scope` needs to be erased from `R`. Unsafely casting the type to remove the `Scope` from the requirement will cause a runtime error (both previously and now)